File: 2025-26/src/lecture03/ackages/Player.cpp
#include <iostream>
using namespace std;
class Player {
    int vita = 10;
public:
    char name = 's';
    void damage(){
        this->vita -= 5;
    }
};
int main() {
    Player* p1 = new Player();
    Player* p2 = new Player();
    p1->name = 'r';
    int* bo = (int*)p1;
    cout << "p1: vita " << *bo << "\n";
    bo++;
    cout << "p1: name " << *bo << "\n";
    bo++;
    cout << "?? " << *bo << "\n";
    bo++;
    cout << "?? " << *bo << "\n";
    bo++;
    cout << "p2: vita " << *bo << "\n";
    p2->damage();
    cout << "p2: vita " << *bo << "\n";
    bo++;
    cout << "p2: name " << *bo << "\n";
}

File: 2025-26/src/lecture03/ackages/entities/Player.java
package lecture03.ackages.entities;
public class Player {
    public String username;
    private int health = 20;
    private boolean isPoisoned = false;
    int fakeHealth = 10;
    public void setPoisoned(boolean p){
        this.isPoisoned = p;
    }
    public String getUsername(){
        return this.username;
    }
    public void damage(int amount) {
        this.health = this.health - amount;
        if (this.health <= 0) {
            System.out.println("Player died!");
        }
    }
    public void isAlive(){
        if (this.health <= 0 ){
            System.out.println("Giocatore morto");
        }else{
            System.out.println("Giocatore vivo");
        }
    }
    public void poison() {
        this.setPoisoned(true);
    }
    public void poisonDamage(){
        if (!this.isPoisoned){
            return;
        }
        if (this.isPoisoned && this.health < 2 ){
            return;
        }
        this.health -= 1;
    }
}